tools/firely-auth/firely-auth-settings.rst

```diff
@@ -0,0 +1,299 @@
+In that documentation you can also read how to use the ``Https`` setting instead of ``HttpsFromPem`` to use a ``.pfx`` file for your SSL certificate.
+
+.. _firely_auth_settings_server:
+
+Firely Server
+^^^^^^^^^^^^^
+
+Firely Auth hands out SMART on FHIR access tokens to access resources on Firely Server. 
+To make Firely Server known to Firely Auth, fill in the ``FhirServerConfig``:
+
+.. code-block:: json
+
+   "FhirServerConfig": {
+      "Name": "Firely Server",
+      "FHIR_BASE_URL": "http://localhost:4080"
+   },
+
+- ``Name``: This name serves two purposes:
+
+  - It can be added to the token as the value of the ``aud`` (audience) claim, if the client requests so. 
+    To have it accepted by Firely Server, set its ``SmartAuthorizationOptions:Audience`` setting to the same value.
+    (This currently only works for the Authorization Code flow.)
+  - It correlates with the clients allowed to access the token introspection endpoint (see below).
+    Therefore it should match the value of ``TokenIntrospectionConfig:TokenIntrospectionResources:Name``
+
+- ``FHIR_BASE_URL``: A token can have a claim in the form of ``patient=<base>/Patient/123``, to define the compartment the client is restricted to.
+  This url is used as the ``base`` part in that url, and should match the base url of Firely Server, as it is accessed by the client.
+
+.. _firely_auth_settings_tokentypes:
+
+Token types
+^^^^^^^^^^^
+
+Define for each client what type of token it can request.
+
+.. code-block:: json
+
+   "TokenConfig": {
+      "AccessTokenType": {
+          "<ClientId>": "Jwt"
+      }
+  },
+
+- ``<ClientId>`` should match one of the clients defined in ``ClientRegistrationConfig``.
+- The value can be one of ``Jwt`` or ``Reference``. ``Jwt`` means that this client will get self-contained Json Web Tokens.
+  ``Reference`` means that this client will get reference tokens, that refer to the actual token kept in memory by Firely Auth.
+  For more background see :term:`reference token`.
+
+E.g. ``"MySmartApp": "Reference"`` 
+
+.. _firely_auth_settings_keymanagement:
+
+Key management
+^^^^^^^^^^^^^^
+
+.. code-block:: json
+
+  "KeyManagementConfig": {
+      "RSA_Config": {
+          //"RSA_JWK": "<JSON Web Key>", // JSON Web Key of type RSA
+          "SupportedAlgorithms": [
+              "RS256",
+              "RS384",
+              "RS512"
+          ]
+      },
+      "EC_Config": {
+          //"JWK_ES256": "<JSON Web Key>", // JSON Web Key of type EC with crv P-256
+          //"JWK_ES384": "<JSON Web Key>", // JSON Web Key of type EC with crv P-384
+          //"JWK_ES512": "<JSON Web Key>", // JSON Web Key of type EC with crv P-512
+          "SupportedAlgorithms": [
+              "ES256",
+              "ES384",
+              "ES512"
+          ]
+      }
+  }
+
+Firely Auth can work with multiple signature keys, used to sign access tokens. 
```

Comment:
Is it only `access tokens`, not all tokens (`refresh`, `id`)?

Reply:
Indeed. Changed it to "access and other tokens"